src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -859,8 +876,36 @@ def update_agents(cmd, client, resource_group_name, cluster_name, https_proxy=""
     # Setting the config dataplane endpoint
     config_dp_endpoint = get_config_dp_endpoint(cmd, connected_cluster.location)
 
+    if auto_upgrade != "false":
+        helm_values = get_all_helm_values(release_namespace, kube_config, kube_context)
+        auto_upgrade = helm_values.get('systemDefaultValues').get('azureArcAgents').get('autoUpdate')
+        if not auto_upgrade:
+            auto_upgrade = "false"
+
+    if default_registry:
+        container_registry_repository = None
```

Comment:
This scenario should be covered in initial pcr parameter validations. We should anyways throw an error if both container_registry_repository and default_registry are set

Reply:
ack

Reply:
Since, there will be a lot of validations related to pcr parameters, its better to define those checks here -
https://github.com/Azure/azure-cli-extensions/blob/main/src/connectedk8s/azext_connectedk8s/_validators.py